source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/utils/wrappers/torchrl/torchrl_env_wrapper.py

```diff
@@ -0,0 +1,601 @@
+        return tensordict
+
+
+class SyncDataCollectorWrapper(SyncDataCollector):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def iterator(self) -> Iterator[TensorDictBase]:
+        """Iterates through the DataCollector.
+
+        Yields: TensorDictBase objects containing (chunks of) trajectories
+
+        """
+        # The portion of the code handling cuda streams has been removed in this inherited method, which
+        # caused CUDA memory allocation issues with IsaacSim during env stepping.
+        total_frames = self.total_frames
+
+        while self._frames < self.total_frames:
+            self._iter += 1
+            # Log the time taken by self.rollout()
+            start_time = time.perf_counter()
+            tensordict_out = self.rollout()
+            end_time = time.perf_counter()
+            rollout_time = end_time - start_time
+            # Add rollout time to tensordict
+            tensordict_out.set("rollout_time", torch.tensor(rollout_time).expand(tensordict_out.shape))
+            self._frames += tensordict_out.numel()
+            if self._frames >= total_frames:
+                self.env.close()
+
+            if self.split_trajs:
+                tensordict_out = split_trajectories(tensordict_out, prefix="collector")
+            if self.postproc is not None:
+                tensordict_out = self.postproc(tensordict_out)
+            if self._exclude_private_keys:
+
+                def is_private(key):
+                    if isinstance(key, str) and key.startswith("_"):
+                        return True
+                    if isinstance(key, tuple) and any(_key.startswith("_") for _key in key):
+                        return True
+                    return False
+
+                excluded_keys = [key for key in tensordict_out.keys(True) if is_private(key)]
+                tensordict_out = tensordict_out.exclude(*excluded_keys, inplace=True)
+            if self.return_same_td:
+                # This is used with multiprocessed collectors to use the buffers
+                # stored in the tensordict.
+                # if events:
+                #     for event in events:
+                #         event.record()
+                #         event.synchronize()
+                yield tensordict_out
+            else:
+                # we must clone the values, as the tensordict is updated in-place.
+                # otherwise the following code may break:
+                # >>> for i, data in enumerate(collector):
+                # >>>      if i == 0:
+                # >>>          data0 = data
+                # >>>      elif i == 1:
+                # >>>          data1 = data
+                # >>>      else:
+                # >>>          break
+                # >>> assert data0["done"] is not data1["done"]
+                yield tensordict_out.clone()
+
+
+class ClipPPOLossWrapper(ClipPPOLoss):
```

Comment:
Can you comment on what has been changed here?
There are many versions of PPO and we probably don't cover exactly what you want it to do but that can be fixed!